app/institutions/dashboard/route.ts

```diff
@@ -1,21 +1,36 @@
 import { action } from '@ember-decorators/object';
 import { service } from '@ember-decorators/service';
 import Route from '@ember/routing/route';
-import DS from 'ember-data';
 
+import { task } from 'ember-concurrency';
 import Analytics from 'ember-osf-web/services/analytics';
 
 export default class InstitutionsDashboardRoute extends Route {
     @service analytics!: Analytics;
-    @service store!: DS.Store;
+    @service router!: any;
 
-    // eslint-disable-next-line camelcase
-    model(params: { institution_id: string }) {
-        return this.store.findRecord('institution', params.institution_id);
+    loadInstitutionsDashboardRoute = task(function *(this: InstitutionsDashboardRoute, institutionId: string) {
+        try {
+            const institution = yield this.get('store').findRecord('institution', institutionId);
+            if (!institution.get('currentUserIsAdmin')) {
+                throw new Error('Current user is not admin.');
+            }
+            return institution;
+        } catch (error) {
+            this.transitionTo('not-found', this.get('router').get('currentURL').slice(1));
+            return undefined;
+        }
+    });
+
+    // TODO Will need to check user permission to access institutional dashboard
+    model(this: InstitutionsDashboardRoute, params: Record<string, string>) {
```

Comment:
Typing `this` isn't required here, we have this in a number of other routes that we haven't gone back and cleaned up, but it's not necessary anymore.